src/server/ftpserver.rs

```diff
@@ -456,32 +456,25 @@ where
         // 3. put expiry time in the LIFO list
         // 4. send reply to client: "Entering Passive Mode ({},{},{},{},{},{})"
 
-        let mut p1 = 0;
-        let mut p2 = 0;
+        let mut reserved_port: u16 = 0;
         if let Some(switchboard) = &mut self.proxy_protocol_switchboard {
             let port = switchboard.reserve_next_free_port(session_arc.clone()).await.unwrap();
             slog::info!(self.logger, "Reserving data port: {:?}", port);
-            p1 = port >> 8;
-            p2 = port - (p1 * 256);
+            reserved_port = port
         }
         let session = session_arc.lock().await;
         if let Some(conn) = session.control_connection_info {
-            let octets = match self.passive_host {
-                PassiveHost::IP(ip) => ip.octets(),
-                PassiveHost::FromConnection => match conn.to_ip {
-                    IpAddr::V4(ip) => ip.octets(),
-                    IpAddr::V6(_) => panic!("Won't happen."),
-                },
+            let conn_ip = match conn.to_ip {
+                IpAddr::V4(ref ip) => ip,
+                IpAddr::V6(_) => panic!("Won't happen."),
```

Comment:
Maybe elaborate why won't it happen, in case it happens :).

Reply:
Yeah OK, its because PASV only does IPV4.

Reply:
Added tot panic msg